### ISSUE TITLE
Cannot rename index in migrations

### ISSUE THREAD

**Issue by JHeimbach** [dachcom-digital/pimcore-formbuilder#243]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no

Hello, 
i'm trying to update a pimcore v5.8 system to v6.7.2 and formbuilder from 2.8 to 3.3. everything works fine locally but on update to our server i get an error on the migrations: it can't rename the unique constraint from `name` to `UNIQ_29DA5346999517A`

our mysql server is: MySQL: 5.5.5-10.2.21-MariaDB-log

the migration command returns this error message:
```

Migrating up to 20200502174518 from 20190424141133
  ++ migrating 20200204093628
Migration 20200204093628 was executed but did not result in any SQL statements.
  ++ migrated (0.63s)
  ++ migrating 20200304172137
     -> ALTER TABLE formbuilder_forms ENGINE=InnoDB;
     -> CREATE TABLE formbuilder_output_workflow_channel (id INT AUTO_INCREMENT NOT NULL, output_workflow INT DEFAULT NULL, type VARCHAR(190) NOT NULL, configuration LONGTEXT DEFAULT NULL COMMENT "(DC2Type:object)", INDEX IDX_CEC462362C75DDDC (output_workflow), PRIMARY KEY(id)) DEFAULT CHARACTER SET UTF8MB4 COLLATE `utf8mb4_general_ci` ENGINE = InnoDB ROW_FORMAT=DYNAMIC;
     -> CREATE TABLE formbuilder_output_workflow (id INT AUTO_INCREMENT NOT NULL, form_definition INT DEFAULT NULL, `name` VARCHAR(255) DEFAULT NULL, success_management LONGTEXT DEFAULT NULL COMMENT "(DC2Type:object)", INDEX IDX_BCB7909761F7634C (form_definition), UNIQUE INDEX name_form (name, form_definition), PRIMARY KEY(id)) DEFAULT CHARACTER SET UTF8MB4 COLLATE `utf8mb4_general_ci` ENGINE = InnoDB ROW_FORMAT=DYNAMIC;
     -> ALTER TABLE formbuilder_output_workflow_channel ADD CONSTRAINT FK_CEC462362C75DDDC FOREIGN KEY (output_workflow) REFERENCES formbuilder_output_workflow (id);
     -> ALTER TABLE formbuilder_output_workflow ADD CONSTRAINT FK_BCB7909761F7634C FOREIGN KEY (form_definition) REFERENCES formbuilder_forms (id);
     -> ALTER TABLE formbuilder_forms CHANGE mailLayout mailLayout LONGTEXT DEFAULT NULL COMMENT "(DC2Type:object)";
     -> ALTER TABLE formbuilder_forms RENAME INDEX name TO UNIQ_29DA5346999517A;
Migration 20200304172137 failed during Execution. Error An exception occurred while executing 'ALTER TABLE formbuilder_forms RENAME INDEX name TO UNIQ_29DA5346999517A;':
SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'INDEX name TO UNIQ_29DA5346999517A' at line 1
```

my google search concluded that mariadb 10.2 has no rename command for indices (or constraints for that matter), but i'm not sure if this is correct thats why i created this ticket and not a pull request.

I think we can fix this with the following change, but i'm not sure about mariadb 10.0 and mysql 5.6 (pimcores minimun supported mysql versions). maybe someone here has a more profound mysql-version knowledge.

```diff
class Version20200304172137 extends AbstractPimcoreMigration implements ContainerAwareInterface
{
    // ...
    public function up(Schema $schema)
    {
        // ...
-        $this->addSql('ALTER TABLE formbuilder_forms RENAME INDEX name TO UNIQ_29DA5346999517A;');
+        $this->addSql('ALTER TABLE formbuilder_forms DROP CONSTRAINT `name`, ADD CONSTRAINT UNIQ_29DA5346999517A UNIQUE (`name`);');

       // ...
    }
```

**Comment by solverat** [comment]
We could adjust the rename like this:

```sql
ALTER TABLE formbuilder_forms DROP INDEX `name`;
ALTER TABLE formbuilder_forms ADD CONSTRAINT UNIQ_29DA5346999517A UNIQUE (`name`);
```

WDYT?

**Comment by JHeimbach** [comment]
Works for me, i'll create a Pull request

**Comment by solverat** [comment]
fixed with #247.
